services/audio/src/main/java/ca/bc/gov/tno/services/audio/handlers/FetchDataService.java

```diff
@@ -0,0 +1,202 @@
+package ca.bc.gov.tno.services.audio.handlers;
+
+import java.time.LocalDateTime;
+import java.io.File;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.EventListener;
+import ca.bc.gov.tno.services.data.events.TransactionBeginEvent;
+import ca.bc.gov.tno.services.events.ErrorEvent;
+import ca.bc.gov.tno.services.audio.config.AudioConfig;
+import ca.bc.gov.tno.services.audio.events.ProducerSendEvent;
+
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+/**
+ * FetchDataService class, provides an event handler that will make requests for
+ * data sources when a FetchEvent is fired.
+ */
+@Async
+@Component
+public class FetchDataService implements ApplicationListener<TransactionBeginEvent<AudioConfig>> {
+  private static final Logger logger = LogManager.getLogger(FetchDataService.class);
+
+  private final ApplicationEventPublisher eventPublisher;
+
+  private Object caller;
+  private AudioConfig audioConfig;
+
+  /**
+   * Create a new instance of a FetchDataService object.
+   */
+  @Autowired
+  public FetchDataService(final ApplicationEventPublisher eventPublisher) {
+    this.eventPublisher = eventPublisher;
+  }
+
+  /**
+   * Start an audio stream, if not already in progress, and extract an audio clip. Publish an audio event so that the Kafka 
+   * Producer can push the content to a topic.
+   */
+  @Override
+  @EventListener
+  public void onApplicationEvent(TransactionBeginEvent<AudioConfig> event) {
+
+    audioConfig = event.getConfig();
+    String mediaSource = audioConfig.getCaptureDir() + "/" + audioConfig.getId() + ".mpg";
+    String cmd = audioConfig.getClipCmd();
+    String destination = clipPath();
+
+    try {
+      caller = event.getSource();
+      var url = audioConfig.getAudioUrl();
+      var readyEvent = new ProducerSendEvent(caller, audioConfig, destination);
+
+      if (startAudioStream()) {
+        eventPublisher.publishEvent(readyEvent);
+      } else {
+        executeClipCmd(cmd, mediaSource, destination);
+        eventPublisher.publishEvent(readyEvent);
+      }
+    } catch (Exception ex) {
+      var errorEvent = new ErrorEvent(this, ex);
+      eventPublisher.publishEvent(errorEvent);
+    }
+  }
+
+  /**
+   * Obtain the location of the streaming file for this event and check that it is being written to.
+   * If the file doesn't exist, or the audio stream has dropped, start streaming from the media source.
+   */
+  private boolean startAudioStream() {
+
+    String captureFilePath = audioConfig.getCaptureDir() + "/" + audioConfig.getId() + ".mpg";
+    File captureFile = new File(captureFilePath);
```

Comment:
java.io.File is just an abstract representation of a file path rather than an open file that you can read. The file doesn't require closing and will be garbage collected after startAudioStream() is popped off the stack.